include/fiction/utils/layout_utils.hpp

```diff
@@ -687,6 +688,105 @@ template <typename CoordinateType>
 }
 #pragma GCC diagnostic pop
 
+/**
+ * This function checks whether the given layouts `first_lyt` and `second_lyt` are identical by comparing various
+ * properties such as the number of cells, the types of cells, defects (if applicable), and charge states (if
+ * applicable). The comparison is done in a detailed manner depending on the specific layout type.
+ *
+ * @tparam Lyt The layout type. Must be a cell-level layout.
+ * @param first_lyt The first layout to compare.
+ * @param second_lyt The second layout to compare.
+ * @return `true` if the layouts are identical, `false` otherwise.
+ */
+template <typename Lyt>
+[[nodiscard]] inline bool are_layouts_identical(const Lyt& first_lyt, const Lyt& second_lyt) noexcept
```

Comment:
Since this compares cell-level layouts particularly, let's maybe rename it to `are_cell_layouts_identical`